test/Cardano/Address/Style/ShelleySpec.hs

```diff
@@ -725,16 +847,119 @@ prop_roundtripEnumAccountingStyle :: AccountingStyle -> Property
 prop_roundtripEnumAccountingStyle ix =
     (toEnum . fromEnum) ix === ix
 
+prop_pointerAddressConstruction
+    :: (SomeMnemonic, SndFactor)
+    -> AccountingStyle
+    -> Index 'Soft 'AddressK
+    -> NetworkDiscriminant Shelley
+    -> ChainPointer
+    -> Property
+prop_pointerAddressConstruction (mw, (SndFactor sndFactor)) cc ix net ptr =
+    pointerAddr `seq` property ()
+  where
+    rootXPrv = genMasterKeyFromMnemonic mw sndFactor :: Shelley 'RootK XPrv
+    accXPrv  = deriveAccountPrivateKey rootXPrv minBound
+    addrXPub = toXPub <$> deriveAddressPrivateKey accXPrv cc ix
+    pointerAddr = pointerAddress net addrXPub ptr
+
+
 {-------------------------------------------------------------------------------
                              Golden tests
 -------------------------------------------------------------------------------}
+-- | Definitions: Let's assume we have private key         < xprv | pub | cc >
+-- Then, extended private key should be understood as      < xprv |     | cc >
+--       extended public key  should be understood as      <      | pub | cc >
+--       verification public key should be understood as   <      | pub |    >
 
-data TestVector = TestVector
+data GoldenTestPointerAddress = GoldenTestPointerAddress
     {
-      -- | Definitions: Let's assume we have private key  < xprv | pub | cc >
-      -- Then, extended private key is understood as < xprv |     | cc >
-      --       extended public key  is understood as <      | pub | cc >
+      -- | The verification public key as string to be encoded into base16 form
+      -- After that it should have 32-byte length
+       verKey :: Text
+
+      -- | Location of stake cerificate in the blockchain
+    ,  stakePtr :: ChainPointer
+
+      -- | Network id
+    ,  networkId :: Word8
 
+      -- | Expected address as encoded into base16 form
+    ,  expectedAddr :: ByteString
+    }
+
+goldenTestPointerAddress :: GoldenTestPointerAddress -> SpecWith ()
+goldenTestPointerAddress GoldenTestPointerAddress{..} =
+    it ("pointer address for networkId " <> show networkId) $ do
+
+    let (Just xPub) = xpubFromBytes $ b16encode $ T.append verKey verKey
+    let addrXPub = liftXPub xPub :: Shelley 'AddressK XPub
+    let (Right tag) = mkNetworkDiscriminant networkId
+    let ptrAddr = pointerAddress tag addrXPub stakePtr
+    let (Right expectedAddr') = fromBase16 expectedAddr
```

Comment:
indentation is a bit misleading here.

Reply:
corrected in https://github.com/input-output-hk/cardano-addresses/pull/31/commits/1779866d826ce736ee88893d3dd4ac2fadd9e1af